anvio/kegg.py

```diff
@@ -0,0 +1,2346 @@
+
+class KeggSetup(KeggContext):
+    """Class for setting up KEGG Kofam HMM profiles and modules.
+
+    It performs sanity checks and downloads, unpacks, and prepares the profiles for later use by `hmmscan`.
+    It also downloads module files and creates the MODULES.db.
+
+    Parameters
+    ==========
+    args: Namespace object
+        All the arguments supplied by user to anvi-setup-kegg-kofams
+    """
+
+    def __init__(self, args, run=run, progress=progress):
+        self.args = args
+        self.run = run
+        self.progress = progress
+
+        # init the base class
+        KeggContext.__init__(self, self.args)
+
+        filesnpaths.is_program_exists('hmmpress')
+
+        if not args.reset and not anvio.DEBUG:
+            self.is_database_exists()
+
+        filesnpaths.gen_output_directory(self.kegg_data_dir, delete_if_exists=args.reset)
+        filesnpaths.gen_output_directory(self.hmm_data_dir, delete_if_exists=args.reset)
+        filesnpaths.gen_output_directory(self.orphan_data_dir, delete_if_exists=args.reset)
+        filesnpaths.gen_output_directory(self.module_data_dir, delete_if_exists=args.reset)
+        filesnpaths.gen_output_directory(self.pathway_data_dir, delete_if_exists=args.reset)
+
+        # ftp path for HMM profiles and KO list
+            # for ko list, add /ko_list.gz to end of url
+            # for profiles, add /profiles.tar.gz  to end of url
+        self.database_url = "ftp://ftp.genome.jp/pub/db/kofam"
+        # dictionary mapping downloaded file name to final decompressed file name or folder location
+        self.files = {'ko_list.gz': self.ko_list_file_path, 'profiles.tar.gz': self.kegg_data_dir}
+
+        # Kegg module text files
+        self.kegg_module_download_path = "https://www.genome.jp/kegg-bin/download_htext?htext=ko00002.keg&format=htext&filedir="
+        self.kegg_pathway_download_path = "https://www.genome.jp/kegg-bin/download_htext?htext=br08901.keg&format=htext&filedir="
+        self.kegg_rest_api_get = "http://rest.kegg.jp/get"
+
+
+    def is_database_exists(self):
+        """This function determines whether the user has already downloaded the Kofam HMM profiles and KEGG modules."""
+
+        if os.path.exists(self.kofam_hmm_file_path):
+            raise ConfigError("It seems you already have KOfam HMM profiles installed in '%s', please use --reset flag if you want to re-download it." % self.kegg_data_dir)
+
+        if os.path.exists(self.kegg_module_file):
+            raise ConfigError("Interestingly, though KOfam HMM profiles are not installed on your system, KEGG module information seems to have been \
+            already downloaded in %s. Please use the --reset flag to re-download everything from scratch." % self.kegg_data_dir)
+
+        if os.path.exists(self.kegg_pathway_file):
+            raise ConfigError("Interestingly, though KOfam HMM profiles are not installed on your system, KEGG pathway information seems to have been \
```

Comment:
we don't care too much about line length in general, but these lines are exceptionally long. plus all these raise statements are missing alignments and use `\` for new lines. I also saw other `run.warnings` statements that suffered from similar things :) I will not point out everything, but please take another look at your modules.